### PR TITLE
Add draft Service Continuity Plan document + blog post

### PR DESCRIPTION
**Pull request by javierpena**
This will allow for a broader discussion.

### CODE REVIEW

**Comment by javierpena** [comment]
Oh, I didn't see that the PR wasn't merged, so there's two different changes here:

- Service Continuity draft document
- Blog post

File: source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  * Puppet and Ansible as configuration management and orchestration tools
+- **Development**
+  * Git as a code repository
+  * Gerrit as a code review and repository management tool
+  * Etherpad as a collaborative editing tool
+- **Continuous integration**
+  * Zuul as an orchestrator of the gate checks
+  * Nodepool to automate the creation and deletion of short-lived VMs for CI jobs across multiple clouds
+  * Jenkins to execute CI jobs (actually, it has now been replaced by Zuul itself)
+
```
Comment (amoralej) [inline source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md]: I'd add Jenkins Job Builder to define CI jobs as code also.

File: source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+- **Development practices**: each OpenStack project follows the same structure:
+
+  * There is a Project Team Leader (PTL), elected from the project contributors every six months. A PTL acts as a project coordinator, rather than a manager in the traditional sense, and is usually expected to rotate every few cycles.
+  * There are several core reviewers, people with enough knowledge on the project to judge if a change is correct or not.
+  * And then we have multiple project contributors, who can create patches and peer-review other people's patches.
+
+  Whenever a patch is created, it is sent to review using a code review system, and then:
+
+  * It is checked by multiple CI jobs, that ensure the patch is not breaking any existing functionality.
+  * It is reviewed by other contributors.
```
Comment (amoralej) [inline source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md]: I'd reinforce the importance of peer review process by explaining its advantages, as enforcing people justify and document changes, enforce some policies, it's very good for newcomers, etc... just an idea, you could create subsections for peer-review and gating CI with a bit longer explanation.

File: source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+---
+title: Some lessons an IT department can learn from OpenStack
+author: jpena
+tags: 
+date: 2017-05-17 13:39:19 CEST
+---
+
+I have spent a lot of my professional carrer working as an IT Consultant/Architect. In those positions, you talk to many customers with different backgrounds, and see companies that run their IT in many different ways. Back in 2014, I joined the OpenStack Engineering team at Red Hat, and started being involved with the OpenStack community. And guess what, I found yet another way of managing IT.
```
Comment (chkumar246) [inline source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md]: s/carrer/career

File: source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md (hunk 1 above)
Comment (chkumar246) [inline source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md]: Tempest is also used as an integration and validation suite. Do we need to mention about it?

File: source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## Processes
+
+The OpenStack community relies on several processes to develop a _cloud operating system_. Most of these processes have evolved over time, and together they allow a very large contributor base to collaborate effectively. Also, we need to manage a complex infrastructure to support this our processes.
+
+- **Infrastructure as code**: there are several important servers in the OpenStack infrastructure, providing service to thousands of users every day: the Git repositories, the Gerrit code review infrastructure, the CI bits, etc. The deployment and configuration of all those pieces is automated, as you would expect, and the Puppet modules and Ansible playbooks used to do so are available at their [Git repository](https://github.com/openstack-infra/system-config). There can be no snowflakes, no "this server requires a very specific configuration, so I have to log on and do it manually" excuses. If it cannot be automated, it is not efficient enough. Also, storing our infrastructure definitions as code allows us to take changes through peer-review and CI before applying in production. More about that later.
```
Comment (chkumar246) [inline source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md]: s/https://github.com/openstack-infra/system-config/http://git.openstack.org/cgit/openstack-infra/system-config

File: source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+  Whenever a patch is created, it is sent to review using a code review system, and then:
+
+  * It is checked by multiple CI jobs, that ensure the patch is not breaking any existing functionality.
+  * It is reviewed by other contributors.
+
+  When the patch passes all CI jobs, and received enough positive votes from the core reviewers, it goes through another round of CI jobs and is finally merged in the repository. This may seem as a complex process, but it has several advantages:
+
+  * It ensures a certain level of quality on the master branch, since we have to ensure that CI jobs are passing.
+  * In encourages peer reviews, so code should always be checked by more than one person before merging.
```
Comment (chkumar246) [inline source/blog/2017-05-17-some-lessons-an-it-department-can-learn-from-openstack.html.md]: S/In/It

**Comment by javierpena** [comment]
Thanks for the comments, I have updated the post.

**Comment by duck-rh** [comment]
All fine for me, let's the community be aware of it now :-)

**Pull request merged**